### PR TITLE
IRS Demo - split IRS Demo into two separate applications to showcase …

### PR DESCRIPTION
**Pull request by m4ksio**
…the separation and usage of RPC client

### CODE REVIEW

File: samples/irs-demo/build.gradle [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+buildscript {
+    ext {
+        springBootVersion = '1.5.6.RELEASE'
+    }
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
+        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${kotlin_version}")
```
Comment (Clintonio) [inline samples/irs-demo/build.gradle]: Are these two lines needed? They should be on the classpath from the root build.gradle.

File: samples/irs-demo/build.gradle [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
-apply plugin: 'maven-publish'
-apply plugin: 'application'
+apply plugin: 'org.springframework.boot'
 
-mainClassName = 'net.corda.irs.IRSDemo'
+buildscript {
+    ext {
+        springBootVersion = '1.5.6.RELEASE'
+    }
+    repositories {
```
Comment (Clintonio) [inline samples/irs-demo/build.gradle]: Is this needed or does it get inherited from the root?

File: samples/irs-demo/build.gradle [hunk 3]
```diff
@@ -22,99 +30,43 @@ sourceSets {
 configurations {
     integrationTestCompile.extendsFrom testCompile
     integrationTestRuntime.extendsFrom testRuntime
+    demoArtifacts.extendsFrom testRuntime
+    all*.exclude module : 'spring-boot-starter-logging'
 }
 
+ext['artemis.version'] = "2.1.0"
```
Comment (Clintonio) [inline samples/irs-demo/build.gradle]: What is the purpose of these two lines?

File: samples/irs-demo/cordapp/build.gradle [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    // Specify your cordapp's dependencies below, including dependent cordapps
+    compile group: 'commons-io', name: 'commons-io', version: '2.5'
+
+    testCompile project(':node-driver')
+    testCompile "junit:junit:$junit_version"
+    testCompile "org.assertj:assertj-core:${assertj_version}"
+}
+
+
+
```
Comment (Clintonio) [inline samples/irs-demo/cordapp/build.gradle]: Excess whitespace

File: samples/irs-demo/cordapp/build.gradle [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+idea {
+    module {
+        downloadJavadoc = true // defaults to false
+        downloadSources = true
+    }
+}
+
+publishing {
```
Comment (Clintonio) [inline samples/irs-demo/cordapp/build.gradle]: This should already be applied from the publish-utils plugin.

File: samples/irs-demo/web/build.gradle [hunk 6]
```diff
@@ -0,0 +1,75 @@
+buildscript {
+	ext {
+		springBootVersion = '1.5.6.RELEASE'
+	}
+	repositories {
+		mavenCentral()
+	}
+	dependencies {
+		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
+		classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${kotlin_version}")
```
Comment (Clintonio) [inline samples/irs-demo/web/build.gradle]: Same as above

File: samples/irs-demo/web/build.gradle [hunk 7]
```diff
@@ -0,0 +1,75 @@
+buildscript {
+	ext {
+		springBootVersion = '1.5.6.RELEASE'
+	}
+	repositories {
```
Comment (Clintonio) [inline samples/irs-demo/web/build.gradle]: Same as above

File: samples/irs-demo/web/build.gradle [hunk 8]
```diff
@@ -0,0 +1,75 @@
+buildscript {
+	ext {
+		springBootVersion = '1.5.6.RELEASE'
```
Comment (Clintonio) [inline samples/irs-demo/web/build.gradle]: This is common between the two - can it go in the root IRS project?

File: samples/irs-demo/web/build.gradle [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+apply plugin: 'kotlin'
+apply plugin: 'kotlin-spring'
+apply plugin: 'eclipse'
+apply plugin: 'org.springframework.boot'
+apply plugin: 'project-report'
+apply plugin: 'application'
+
+version = '0.0.1-SNAPSHOT'
```
Comment (Clintonio) [inline samples/irs-demo/web/build.gradle]: Please remove - it should track the corda version for now.

File: samples/irs-demo/web/build.gradle [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+sourceCompatibility = 1.8
+compileKotlin {
```
Comment (Clintonio) [inline samples/irs-demo/web/build.gradle]: These are applied in the all projects section of the root build.gradle

File: samples/irs-demo/web/build.gradle [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+	kotlinOptions.jvmTarget = "1.8"
+}
+compileTestKotlin {
```
Comment (Clintonio) [inline samples/irs-demo/web/build.gradle]: This too These are applied in the all projects section of the root build.gradle

File: samples/irs-demo/web/build.gradle [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+	kotlinOptions.jvmTarget = "1.8"
+}
+
+repositories {
+	mavenCentral()
```
Comment (Clintonio) [inline samples/irs-demo/web/build.gradle]: This is also applied from the root project.

File: samples/irs-demo/web/build.gradle [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+// Spring declares dependency on old Artemis Client 1.5.5 which is incompatible with Corda
+// it's fixed in Spring boot starter 2.0 but it's not released officially at the time of writing this.
+ext['artemis.version'] = "2.1.0"
```
Comment (Clintonio) [inline samples/irs-demo/web/build.gradle]: What is the purpose of these two lines?

File: samples/irs-demo/web/build.gradle [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+		exclude module: "logback-classic"
+	}
+}
+
+jar {
+	from sourceSets.test.output
+}
+
+task deployWebapps(type: Copy, dependsOn: ['jar', 'bootRepackage']) {
+
```
Comment (Clintonio) [inline samples/irs-demo/web/build.gradle]: -whitespace

File: samples/network-visualiser/build.gradle [hunk 15]
```diff
@@ -1,15 +1,30 @@
+buildscript {
```
[the remark this answers is not in the document]
Comment (m4ksio) [reply samples/network-visualiser/build.gradle]: Unfortunately yes, due to nature of Spring Boot Gradle pluign, as noted in the build file itself.

**Pull request merged**